algorithms/unit_tests/TestSortByKey.hpp

```diff
@@ -24,10 +24,29 @@
 
 #include <utility>  // pair
 
+#if defined(KOKKOS_ENABLE_ONEDPL)
+#define KOKKOS_IMPL_ONEDPL_VERSION                            \
+  ONEDPL_VERSION_MAJOR * 10000 + ONEDPL_VERSION_MINOR * 100 + \
+      ONEDPL_VERSION_PATCH
+#define KOKKOS_IMPL_ONEDPL_VERSION_GREATER_EQUAL(MAJOR, MINOR, PATCH) \
+  (KOKKOS_IMPL_ONEDPL_VERSION >= ((MAJOR)*10000 + (MINOR)*100 + (PATCH)))
+#endif
+
+#ifndef KOKKOS_IMPL_ONEDPL_VERSION_GREATER_EQUAL
+#define KOKKOS_IMPL_ONEDPL_VERSION_GREATER_EQUAL(MAJOR, MINOR, PATH) 0
+#endif
```

Comment:
We might just consolidate all these macros into a separate header that is included in the implementations and the tests in a separate pull request (and do the same for HIP and Cuda).